bcos-ledger/src/libledger/Ledger.cpp

```diff
@@ -267,29 +344,66 @@ void Ledger::asyncPrewriteBlock(bcos::storage::StorageInterface::Ptr storage,
         LEDGER_LOG(WARNING) << "Empty transactions and metadata, empty block?"
                             << LOG_KV("blockNumber", blockNumberStr);
     }
-    bytes transactionsBuffer;
-    transactionsBlock->encode(transactionsBuffer);
 
-    Entry number2TransactionHashesEntry;
-    number2TransactionHashesEntry.importFields({std::move(transactionsBuffer)});
-    storage->asyncSetRow(SYS_NUMBER_2_TXS, blockNumberStr, std::move(number2TransactionHashesEntry),
-        [setRowCallback](auto&& error) { setRowCallback(std::forward<decltype(error)>(error)); });
+    if (blockVersion >= uint32_t(bcos::protocol::BlockVersion::V3_6_VERSION))
+    {
+        // The genesis block does not have any txs,
+        // so it need to be stored separately in the SYS_NUMBER_2_BLOCK_TXS table.
+        // if (header->number() == SYS_CONTRACT_DEPLOY_NUMBER)
+        if (block->transactionsSize() == 0 || block->transactionsMetaDataSize() == 0)
+        {
+            bytes blockBuffer;
```

Comment:
只有创世块交易为空，用assert去限制，比较容易发现问题
```suggestion
        if (block->number() == 0)
        {
            assert( block->transactionsSize() == 0);
            assert( block->transactionsMetaDataSize() == 0);
        
            bytes blockBuffer;
```